### PR TITLE
Run ESLint and Prettier linting in worker threads

### PR DESCRIPTION
**Pull request by 72636c**
This gets us from

```
skuba process
ESLint process
Prettier process
tsc process
```

to

```
skuba process
├─ ESLint worker thread
╰─ Prettier worker thread
tsc process
```

---

Worker threads are stable in all Node.js versions we support: https://nodejs.org/docs/latest-v12.x/api/worker_threads.html#worker_threads_worker_threads

---

TODO:

- [x] Changeset
- [x] Integration tests
- [x] Concurrency limiter

### CODE REVIEW

**Comment by 72636c** [comment]
This PR is pretty chunky; I should be able to split it up once we're good with #539.

File: src/cli/lint/external.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  const { eslintOk, prettierOk, tscOk } = await lint(input);
+
+  log.newline();
+
+  if (eslintOk && prettierOk && tscOk) {
+    return;
+  }
+
+  // Some stdio doesn't resolve synchronously, so wait for a bit.
+  await new Promise((resolve) => setTimeout(resolve, 1000));
```
Comment (72636c) [inline src/cli/lint/external.ts:42]:
```suggestion
```

File: src/cli/lint.int.test.ts [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+  args: string[];
+  base: string;
+  exitCode: number | undefined;
+}
+
+test.each`
+  description     | args           | base           | exitCode
+  ${'fixable'}    | ${[]}          | ${'fixable'}   | ${1}
+  ${'ok'}         | ${[]}          | ${'ok'}        | ${1}
+  ${'ok --debug'} | ${['--debug']} | ${'ok'}        | ${1}
```
Comment (72636c) [inline src/cli/lint.int.test.ts:84]: These exit codes are wrong 🤔

File: src/cli/lint/eslint.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+import chalk from 'chalk';
+
+import { createLogger } from '../../utils/logging';
+import { execWorkerThread, startWorkerThread } from '../../utils/worker';
+import { runESLint } from '../adapter/eslint';
+
+import type { Input } from './types';
+
+export const runESLintInMainThread = ({ debug }: Input) =>
```
Comment (etaoins) [inline src/cli/lint/eslint.ts:12]: It's a bit confusing that `runESLintInWorkerThread` calls `runESLintInMainThread`. Maybe `runESlintInCurrentThread` would be more accurate?

**Review by etaoins — APPROVED** [review]
🏆

**Pull request merged**